### PR TITLE
Sonar fixes

### PR DESCRIPTION
**Pull request by acsbendi**
Fixes for all the bugs discovered by sonar.

### CODE REVIEW

**Comment by tomcsanyigergely** [comment]
Okay.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (Lestinon) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: OK!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (Lestinon) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: Good job.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -14,6 +14,8 @@
 
   private int torpedoCount = 0;
 
+  private Random generator = new Random();
```
Comment (Lestinon) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Well done!

**Review by Lestinon — CHANGES_REQUESTED** [review]
Ok, but you forgot about one important thing.

**Review by Lestinon — APPROVED** [review]
Okay.

**Pull request merged**